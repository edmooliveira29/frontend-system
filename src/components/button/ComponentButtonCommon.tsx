import * as React from 'react'
import { styled } from '@mui/material/styles'
import Button, { type ButtonProps } from '@mui/material/Button'
import Stack from '@mui/material/Stack'

const ColorButton = styled(Button)<ButtonProps>(() => ({
  backgroundColor: '#001a32',
  boxShadow: '2px -2px 10px gray',
  margin: '0px 10px',
  fontSize: 16,
  '&:hover': {
    backgroundColor: '#001a32',
    boxShadow: '2px -2px 10px gray',
  },
}))

export function ComponentButtonCommon(props: { text: string; sizeWidth?: string; sizeHeight?: string; loading?: boolean, onClick?:any }) {
  return (
    <Stack spacing={2} direction='row' onClick={props.onClick} >
      <ColorButton
        id='button'
        variant='contained'
        style={{ width: props.sizeWidth ?? '300px', height: props.sizeHeight ?? '40px', margin: '0px 10px' }}
        type='submit'
        disabled={props.loading}
      >
        {props.loading ? <div className="spinner-border text-light" role="status" /> : props.text}
      </ColorButton>
    </Stack>
  )
}
