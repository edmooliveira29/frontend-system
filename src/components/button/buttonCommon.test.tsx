import React from 'react';
import {cleanup, render, screen} from '@testing-library/react';
import {ComponentButtonCommon} from './ComponentButtonCommon';

describe('ComponentButtonCommon', () => {
	afterEach(() => {
		cleanup();
	});

	test('Should renders button with text', () => {
		const buttonText = 'Click me!';
		render(<ComponentButtonCommon text={buttonText} />);
		const buttonElement = screen.getByText(buttonText);
		expect(buttonElement).toMatchSnapshot();
	});

	test('Should renders button with custom width', () => {
		const buttonText = 'Click me!';
		const buttonWidth = '200px';
		render(<ComponentButtonCommon text={buttonText} width={buttonWidth} />);
		const buttonElement = screen.getByText(buttonText);
		expect(buttonElement).toMatchSnapshot();
		expect(buttonElement).toHaveStyle(`width: ${buttonWidth}`);
	});

	test('Should applies styles to button on hover', () => {
		const buttonText = 'Click me!';
		render(<ComponentButtonCommon text={buttonText} />);
		const buttonElement = screen.getByText(buttonText);
		buttonElement.dispatchEvent(new MouseEvent('mouseover', {bubbles: true}));
		expect(buttonElement).toHaveStyle('opacity: 0.8');
		expect(buttonElement).toHaveStyle('background-color: #001a32');
		expect(buttonElement).toHaveStyle('box-shadow: 2px -2px 10px gray');
	});
});
