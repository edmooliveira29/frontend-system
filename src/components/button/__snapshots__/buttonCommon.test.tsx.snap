// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentButtonCommon Should renders button with custom width 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-rjcf26-MuiButtonBase-root-MuiButton-root"
  style="width: 200px; height: 40px; margin: 0px 10px;"
  tabindex="0"
  type="submit"
>
  Click me!
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`ComponentButtonCommon Should renders button with text 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-rjcf26-MuiButtonBase-root-MuiButton-root"
  style="width: 100%; height: 40px; margin: 0px 10px;"
  tabindex="0"
  type="submit"
>
  Click me!
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
